crates/viewer/re_viewer_context/src/selection_state.rs

```diff
@@ -104,267 +52,19 @@ impl InteractionHighlight {
-
-                // TODO(ab): it is not very meaningful to copy the `StoreId` representation, but
-                // that's the best we can do for now. In the future, we should have URIs for
-                // in-memory recordings, and that's what we should copy here.
-                Item::StoreId(id) => Some((ClipboardTextDesc::StoreId, format!("{id:?}"))),
-
-                Item::DataResult(_, instance_path) | Item::InstancePath(instance_path) => Some((
-                    ClipboardTextDesc::EntityPath,
-                    instance_path.entity_path.to_string(),
-                )),
-                Item::ComponentPath(component_path) => Some((
-                    ClipboardTextDesc::EntityPath,
-                    component_path.entity_path.to_string(),
-                )),
-            })
-            .chunk_by(|(desc, _)| *desc);
-
-        let mut clipboard_text = String::new();
-        let mut content_description = String::new();
-
-        for (desc, entries) in &clipboard_texts_per_type {
-            let entries = entries.map(|(_, text)| text).collect_vec();
-
-            let desc = match desc {
-                ClipboardTextDesc::FilePath => "file path",
-                ClipboardTextDesc::Url => "URL",
-                ClipboardTextDesc::AppId => "app id",
-                ClipboardTextDesc::StoreId => "store id",
-                ClipboardTextDesc::EntityPath => "entity path",
-            };
-            if !content_description.is_empty() {
-                content_description.push_str(", ");
-            }
-            if entries.len() == 1 {
-                content_description.push_str(desc);
-            } else {
-                content_description.push_str(&format!("{desc}s"));
-            }
-
-            let texts = entries.into_iter().join("\n");
-            if !clipboard_text.is_empty() {
-                clipboard_text.push('\n');
-            }
-            clipboard_text.push_str(&texts);
-        }
-
-        if !clipboard_text.is_empty() {
-            re_log::info!(
-                "Copied {content_description} to clipboard:\n{}",
-                &clipboard_text
-            );
-            egui_ctx.copy_text(clipboard_text);
-        }
-    }
-}
-
 /// Selection and hover state.
 ///
 /// Both hover and selection are double buffered:
 /// Changes from one frame are only visible in the next frame.
 #[derive(Default, serde::Deserialize, serde::Serialize)]
 #[serde(default)]
 pub struct ApplicationSelectionState {
-    /// Selection of the previous frame. Read from this.
-    selection_previous_frame: ItemCollection,
+    /// The selected items. Write to this with [`re_global_context::SystemCommand::SetSelection`].
+    selection: ItemCollection,
 
-    /// Selection of the current frame. Write to this.
+    /// Has selection changed since the previous frame?
     #[serde(skip)]
-    selection_this_frame: Mutex<ItemCollection>,
+    selection_changed: bool,
```

Comment:
It feels like we should come up with a better way to check if this has changed now... Which is used to fire off an event dispatcher.

Reply:
I agree it's not very elegant and we'd better off joining the system event processing with the event dispatcher (which AFAIK is today used for in-process (rust & py notebooks)callbacks only). But I wouldn't push this refactor iteration further 🤷